lib/analyzer.rb

```diff
@@ -0,0 +1,52 @@
+class Analyzer
+
+    def initialize
```

Comment:
Since all of your methods are taking the same argument, it might be better design to instead initialize an Analyzer object with this argument of `vehicles`. Then you can populate the instance variables (ivars) in the initialize so line 4 might look like `@vehicle_models = vehicle_models(vehicles)` that way your instance variable's data is populated from the get go so all other methods can simply call on that ivar.